integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -1436,4 +1439,206 @@ private String getHostNameForCurl() throws Exception {
       }
     }
   }
+
+  public int getLoadBalancerWebPort() {
+    return loadBalancerWebPort;
+  }
+
+  /**
+   * Shut down a ms by setting serverStartPolicy to NEVER
+   *
+   * @throws Exception
+   */
+  public void shutdownManagedServerUsingServerStartPolicy(String msName) throws Exception {
+    String cmd =
+        "kubectl patch domain "
+            + domainUid
+            + " -n "
+            + domainNS
+            + " -p '{\"spec\":{\"managedServers\":[{\"serverName\":\""
+            + msName
+            + "\",\"serverStartPolicy\":\"NEVER\"}]}}' --type merge";
+
+    logger.info("command to shutdown managed server <" + msName + "> is: " + cmd);
+
+    ExecResult result = ExecCommand.exec(cmd);
+    if (result.exitValue() != 0) {
+      throw new Exception("FAILURE: command " + cmd + " failed, returned " + result.stderr());
+    }
+    String output = result.stdout().trim();
+    logger.info("output from shutting down managed server:\n" + output);
+
+    TestUtils.checkPodDeleted(domainUid + "-" + msName, domainNS);
+  }
+
+  /**
+   * Restart a ms by setting serverStartPolicy to IF_NEEDED
+   *
+   * @throws Exception
+   */
+  public void restartManagedServerUsingServerStartPolicy(String msName) throws Exception {
+    String cmd =
+        "kubectl patch domain "
+            + domainUid
+            + " -n "
+            + domainNS
+            + " -p '{\"spec\":{\"managedServers\":[{\"serverName\":\""
+            + msName
+            + "\",\"serverStartPolicy\":\"IF_NEEDED\"}]}}' --type merge";
+
+    logger.info("command to restart managed server <" + msName + "> is: " + cmd);
+
+    ExecResult result = ExecCommand.exec(cmd);
+    if (result.exitValue() != 0) {
+      throw new Exception("FAILURE: command " + cmd + " failed, returned " + result.stderr());
+    }
+    String output = result.stdout().trim();
+    logger.info("output from restarting managed server:\n" + output);
+
+    TestUtils.checkPodCreated(domainUid + "-" + msName, domainNS);
+    TestUtils.checkPodReady(domainUid + "-" + msName, domainNS);
+  }
+
+  /**
+   * Run the shell script to build .war file and deploy the App in the admin pod
+   *
+   * @param webappName - Web App Name to be deployed
+   * @param scriptName - a shell script to build .war file and deploy the App in the admin pod
+   * @param archiveExt - archive extention
+   * @param infoDirNames - archive information dir location
+   * @param username - weblogic user name
+   * @param password - weblogc password
+   * @throws Exception
+   */
+  private void callShellScriptToBuildWarDeployAppInPod(
```

Comment:
Looks like you have refactored callShellScriptByExecToPod() and created another method, eventually callShellScriptByExecToPod should be removed when we make changes to build testwebapp ...

Reply:
I wished I were able to reuse callShellScriptByExecToPod :(